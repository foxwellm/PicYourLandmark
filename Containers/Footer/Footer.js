import React, { Component } from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import { Icon } from 'react-native-elements';

export class Footer extends Component {

  render() {
    const { changeCurrentPage } = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.buttonContainer}>
          <TouchableOpacity onPress={() => changeCurrentPage('User profile')}>
            <Icon color={this.props.currentPage === 'User profile' ? "#0b80dc" : "#3c4859"} name="user" type="font-awesome" size={30} />
          </TouchableOpacity>
          <TouchableOpacity onPress={() => changeCurrentPage('Tutorial')}>
            <Icon color={this.props.currentPage === 'Camera' ? "#0b80dc" : "#3c4859"} name="book" type="font-awesome" size={30} />
          </TouchableOpacity>
          <TouchableOpacity onPress={() => changeCurrentPage('Home')} >
            <Icon color={this.props.currentPage === 'Home' ? "#0b80dc" : "#3c4859"} name="map-marker" type="font-awesome" size={30}/>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => changeCurrentPage('Collected landmarks')}>
            <Icon color={this.props.currentPage === 'Collected landmarks' ? "#0b80dc" : "#3c4859"} name="image" type="font-awesome" size={30} />
          </TouchableOpacity>
          <TouchableOpacity onPress={() => changeCurrentPage('Login')}>
            <Icon color={this.props.currentPage === 'Login' ? "#0b80dc" : "#3c4859"} name="sign-in" type="font-awesome" size={30} />
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: .3,
    width: '100%',
    backgroundColor: '#e9e9e9',
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  }
});

export default Footer;