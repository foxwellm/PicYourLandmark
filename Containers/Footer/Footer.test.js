import React from 'react';
import { Footer } from './Footer';
import { shallow } from 'enzyme';
import { TouchableOpacity } from 'react-native';

describe('Footer', () => {
  let wrapper;
  let mockChangeCurrentPage;

  beforeEach(() => {
    mockChangeCurrentPage = jest.fn()
    wrapper = shallow(<Footer changeCurrentPage={mockChangeCurrentPage}/>)
  });

  it('should match the snapshot with the correct data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot with 15% containerHeight if the currentPage is Camera', () => {
    wrapper.setProps({ currentPage: 'Camera'})
    expect(wrapper).toMatchSnapshot()
  });


  it('user button should call changeCurrentPage and pass in the string User Profile', () => {
    wrapper.find(TouchableOpacity).at(0).simulate('press')
    expect(mockChangeCurrentPage).toHaveBeenCalledWith('User profile')
  });

  it('map marker/home button should call changeCurrentPage and pass in the string Home', () => {
    wrapper.find(TouchableOpacity).at(1).simulate('press')
    expect(mockChangeCurrentPage).toHaveBeenCalledWith('Home')
  });

  it('camera button should call changeCurrentPage and pass in the string Camera', () => {
    wrapper.find(TouchableOpacity).at(2).simulate('press')
    expect(mockChangeCurrentPage).toHaveBeenCalledWith('Camera')
  });

  it('image/picture button should call changeCurrentPage and pass in the string Collected landmarks', () => {
    wrapper.find(TouchableOpacity).at(3).simulate('press')
    expect(mockChangeCurrentPage).toHaveBeenCalledWith('Collected landmarks')
  });

  it('sign in button should call changeCurrentPage and pass in the string Login', () => {
    wrapper.find(TouchableOpacity).at(4).simulate('press')
    expect(mockChangeCurrentPage).toHaveBeenCalledWith('Login')
  });
});