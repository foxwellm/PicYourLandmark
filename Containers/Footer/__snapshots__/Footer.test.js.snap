// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match the snapshot with 15% containerHeight if the currentPage is Camera 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#e9e9e9",
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      Object {
        "height": "15%",
      },
    ]
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="user"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="map-marker"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="camera"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="image"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="sign-in"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
  </Component>
</Component>
`;

exports[`Footer should match the snapshot with the correct data passed in 1`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#e9e9e9",
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      Object {
        "height": "8%",
      },
    ]
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="user"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="map-marker"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="camera"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="image"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        color="#3c4859"
        name="sign-in"
        size={30}
        type="font-awesome"
      />
    </TouchableOpacity>
  </Component>
</Component>
`;
